### PR TITLE
AgentCache: Properly compute auto-auth token

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: command/agent.go [hunk 1]
```diff
@@ -343,6 +340,20 @@ func (c *AgentCommand) Run(args []string) int {
 			return 1
 		}
 
+		sinkConfig := &sink.SinkConfig{
```
Comment (briankassouf) [inline command/agent.go]: Could we only set this up if use auto auth is enabled?

File: command/agent/cache/handler.go [hunk 2]
```diff
@@ -14,19 +14,21 @@ import (
...
 	"github.com/hashicorp/vault/logical"
 )
 
-func Handler(ctx context.Context, logger hclog.Logger, proxier Proxier, useAutoAuthToken bool, client *api.Client) http.Handler {
+func Handler(ctx context.Context, logger hclog.Logger, proxier Proxier, useAutoAuthToken bool, inmemSink sink.SinkReader) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		logger.Info("received request", "path", r.URL.Path, "method", r.Method)
 
 		token := r.Header.Get(consts.AuthHeaderName)
+		autoAuthToken := inmemSink.Token()
```
Comment (briankassouf) [inline command/agent/cache/handler.go]: can we have inmemSink = nil if useAutoAuthToken is false?

File: command/agent.go (hunk 1 above)
Comment (vishalnayak) [reply command/agent.go]: Done.

File: command/agent/cache/handler.go (hunk 2 above)
Comment (vishalnayak) [reply command/agent/cache/handler.go]: Covered the nil cases in the pushed up commit.

**Pull request merged**